parser.mojo

```diff
@@ -35,6 +36,12 @@ struct StringSlice(CollectionElement, Stringable, Formattable):
             end = self.size
         start = idxs.start.or_else(0)
         return StringSlice(self.ptr.offset(start), end - start)
+
+    fn as_bytes_span(self) -> Span[UInt8, __lifetime_of(self)]:
+        return Span[UInt8, __lifetime_of(self)](
+            unsafe_ptr=self.ptr,
+            len=self.size,
+        )
```

Comment:
An unrelated question but this code suffered a bit from lifetime-related issues... are these lifetime parametrizations useful enough now to tell Mojo that eg. keep stuff around in memory more consistently ? 

I understand here it's just needed because Spans work like that, but I never figured out how to make the StringSlices here behave in such a way to make sure their lifetime prolonged the lifetime of the parser they originated from (since they depend on that memory)